File: domain/src/main/scala/com/anjunar/technologyspeaks/codemirror/CodeMirrorImage.scala
package com.anjunar.technologyspeaks.codemirror

import com.anjunar.scala.mapper.annotations.PropertyDescriptor
import jakarta.persistence.{Basic, Column, Entity, Lob}

import scala.compiletime.uninitialized

@Entity
class CodeMirrorImage extends AbstractCodeMirrorFile {

  @Basic
  @Lob
  @PropertyDescriptor(title = "Data", writeable = true)
  var data: Array[Byte] = uninitialized

  @Basic
  @PropertyDescriptor(title = "Content Type", writeable = true)
  var contentType : String = uninitialized

}

File: application/src/main/typescript/packages/library/src/components/inputs/codemirror/domain/CodeMirrorImage.ts
import {AbstractCodeMirrorFile} from "./AbstractCodeMirrorFile";
import Basic from "../../../../mapper/annotations/Basic";
import {Entity} from "../../../../mapper";

@Entity("CodeMirrorImage")
export class CodeMirrorImage extends AbstractCodeMirrorFile {

    $type = "CodeMirrorImage"

    @Basic()
    data : string
    
    @Basic()   
    contentType : string

}